### PR TITLE
Show a "Loading ..." view instead of a "No storage-time" view when starting up

### LINKED ISSUE #423: The "Usage" view misleadingly reports "zero storage-time" at startup while real info is being loaded
If you start GridSync aimed at a ZKAP-enabled grid then -- briefly -- while it loads information about ZKAPs that are available, it reports "0 GB-months available".

This can be confusing, perhaps frightening, to users since GB-months are what you pay for when you are using a ZKAP-enabled grid (either monetarily or with some other capital).

Instead, tell the user that the info is being loaded and then show the info once it has been loaded.

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #423 

This is my first attempt at writing pytest tests for Qt application code.  I took a guess at what a nice shape might be.  I'm not sure why I had to add the `QApplication([])` call.  It looks like there are already tests for Qt application code... but they fail (as do my new ones) in my environment without this.

I didn't put a lot of thought into the actual words to show here, I mostly spent my effort figuring out how to get things to show up at the right time at all, so more than happy to have edits suggested for the text.

Also ... I did quite a bit of refactoring before figuring out what was going on here.  I can split that into a separate PR if that would help.

Basically 2bfc26d is the fix for the behavior, 951f343 are the unit tests, and the rest are refactoring that should change no behavior.

### CODE REVIEW

**Comment by exarkun** [comment]
Clearly CI is unhappy with this.  I'm working on further changes to fix that.

**Comment by exarkun** [comment]
Seems like all of the pieces now line up just right now, so maybe it's reasonable to have a human review at this point...

File: gridsync/gui/__init__.py [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
+    unread_messages: List[Tuple] = attr.ib(default=attr.Factory(list))
+
+    welcome_dialog: WelcomeDialog = attr.ib()
+    main_window: MainWindow = attr.ib()
+    preferences_window: PreferencesWindow = attr.ib()
+    systray: SystemTrayIcon = attr.ib()
+    debug_exporter: DebugExporter = attr.ib()
+
+    @welcome_dialog.default
+    def _default_welcome_dialog(self):
```
Comment (crwood) [inline gridsync/gui/__init__.py:30]: A _very_ small nitpick (since the scope of these wrapped methods is also very small) but consider using a consistent naming scheme here; some of these methods follow the naming pattern of `_x_default` while others follow `_default_x` (I have no preference as to which of the two forms is better -- and won't hold up merging this because of something so trivial -- I just wanted to point it out).

File: tests/gui/conftest.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import pytest
+from PyQt5.QtWidgets import QApplication
+from twisted.python.filepath import FilePath
+
+from gridsync.gui import Gui
+from gridsync.preferences import Preferences
+
+# Without this, QWidget() aborts the process with "QWidget: Must construct a
+# QApplication before a QWidget".
+app = QApplication([])
```
Comment (crwood) [inline tests/gui/conftest.py:16]: FWIW, this is only necessary if you're running the `gui` tests in isolation (e.g., via `pytest tests/gui` or similar). If the test suite is run in its entirety (e.g., by running just `pytest` or `pytest tests` in the top level of the source tree), the `pytest-qt` plugin will initialize and use it's own `QApplication` instance throughout the run. I'm not sure why it doesn't work the same way in both instances, however... Anyway, there's no harm in keeping this one here; (Py)Qt will just use whichever `QApplication` instance is available first.

**Review by crwood — APPROVED** [review]
Thanks a lot for this.

It took me a bit to grok this (due to the addition of `attrs` -- which I wasn't terribly familiar with) but this looks good to me and takes an approach that I would more or less as I would expect (i.e., of providing a different default view and displaying the chart only when token amounts can be discovered). We can indeed improve the wording/presentation of the "loading" label later on; the important thing, for now, I think, is to guard against giving users the false impression that they have no storage-time remaining.

Anyway, merging... And thanks again!

File: gridsync/gui/__init__.py (hunk 1 above)
Comment (exarkun) [reply gridsync/gui/__init__.py:30]: Huh.  Yes, I agree consistent naming would be good here.  I was _trying_ to be consistent, in fact, I guess I should have tried harder!
